moped-editor/src/views/projects/projectView/KnackSync.js

```diff
@@ -0,0 +1,217 @@
+import React from "react";
+import { MenuItem, Icon, ListItemIcon, ListItemText } from "@material-ui/core";
+
+import { useMutation } from "@apollo/client";
+
+import { UPDATE_PROJECT_KNACK_ID } from "../../../queries/project";
+
+/**
+ * Entry in project menu drop down to trigger sync to Data Tracker
+ * @return {JSX.Element}
+ * @constructor
+ */
+const KnackSync = React.forwardRef(
+  (
+    {
+      project,
+      closeHandler,
+      snackbarHandler,
+      sceneNumber = 514,
+      viewNumber = 3047,
+      knackApplicationId = "6167314778435d001ea3e7cb",
+    },
+    ref
+  ) => {
+    /**
+     * Function to build the correct Knack URL to interact with based on properties and if there will be an
+     * update or an initial sync.
+     * @returns string
+     */
+    const buildUrl = () => {
+      let url =
+        "https://api.knack.com/v1/pages/scene_" +
+        sceneNumber +
+        "/views/view_" +
+        viewNumber +
+        "/records";
+      if (project.moped_project[0].knack_project_id) {
+        // existing record
+        url = url + "/" + project.moped_project[0].knack_project_id;
+      }
+      return url;
+    };
+
+    /**
+     * Function to determine the HTTP method to use base on if there will be an update or initial post to Knack
+     * @returns string
+     */
+    const getHttpMethod = () => {
+      let method = "POST";
+      if (project.moped_project[0].knack_project_id) {
+        method = "PUT";
+      }
```

Comment:
You could write lines 49-52 as:
```js
const method = project.moped_project[0].knack_project_id ? "PUT" : "POST";
```

Reply:
Oh nvm, I see Sergio made an even better suggestion below along the same lines.

Reply:
@mateoclarke - I agree. Both of these suggestions are on point, and I have adopted them both. Thanks!